streaming/storage/deque_test.go

```diff
@@ -0,0 +1,336 @@
+package storage
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/cube2222/octosql"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+func TestDeque(t *testing.T) {
+	prefix := "test_deque"
+	path := "test"
+
+	db, err := badger.Open(badger.DefaultOptions(path))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer func() {
+		_ = db.Close()
+		_ = os.RemoveAll(path)
+	}()
+
+	store := NewBadgerStorage(db)
+	txn := store.BeginTransaction().WithPrefix([]byte(prefix))
+
+	queue := NewDeque(txn)
+
+	values := []octosql.Value{
+		octosql.MakeInt(0),
+		octosql.MakeInt(1),
+		octosql.MakeInt(2),
+		octosql.MakeInt(3),
+		octosql.MakeInt(4),
+		octosql.MakeInt(5),
+	}
+
+	//test push back and push front
+	for i := 3; i < 6; i++ { //write 3 4 5
+		err := queue.PushBack(&values[i])
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	for i := 2; i >= 0; i-- { // write 2 1 0 so the queue becomes 0 1 2 3 4 5
+		err := queue.PushFront(&values[i])
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	// test if all values are there
+	err = queue.isEqualTo(values)
+	if err != nil {
+		log.Fatal("after push back and push front: ", err)
+	}
+
+	println("PushBack test passed")
+
+	// test the iterators
+	it := queue.GetIterator()
+	areEqual, err := TestDequeIterator(it, values)
+
+	if err != nil {
+		log.Fatal("get iterator: ", err)
+	}
+
+	if !areEqual {
+		log.Fatal("the iterator isn't correct")
+	}
+
+	_ = it.Close()
+
+	it = queue.GetIterator(WithReverse())
+	areEqual, err = TestDequeIterator(it, reverseValues(values))
```

Comment:
This is nice reverse iterator test, could You rewrite TestMapIterator in the same way?
I've done this in hurry and it's pretty dumb

Reply:
Will do